addon/event_dispatcher.js

```diff
@@ -0,0 +1,93 @@
+import Ember from "ember";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+import { getOwner } from "@ember/application";
+import LinkComponent from "@ember/routing/link-component";
+
+export default class HTMLAnchorEventDispatcher extends Ember.EventDispatcher {
+  @service router;
+
+  findAnchorElement(event) {
+    let target = event.target;
+    let element = null;
+
+    do {
+      if (target instanceof HTMLAnchorElement) {
+        element = target;
+        break;
+      }
+
+      target = target.parentNode;
+    } while (target && target.nodeType === 1);
+
+    return element;
+  }
+
+  recognizeAnchorURL(event, element) {
+    if (
+      !(
+        (event.which === undefined || event.which === 1) &&
+        !event.ctrlKey &&
+        !event.metaKey
+      )
+    ) {
+      return null;
+    }
+
+    if (element.getAttribute("target") === "_blank") {
+      return null;
+    }
+
+    if (element.hasAttribute("download")) {
+      return null;
+    }
+
+    if (element.hasAttribute("data-ember-action")) {
+      return null;
+    }
+
+    let view = getOwner(this).lookup("-view-registry:main")[element.id];
+    if (view && view instanceof LinkComponent) {
+      return null;
+    }
+
+    let href = element.getAttribute("href");
+    if (!href || href.indexOf(this.router.rootURL) !== 0) {
+      return null;
+    }
+
+    if (!this.router.recognize(href)) {
```

Comment:
Do we need to strip the rootURL before recognizing? I can't recall the exact API details here, have you tested this with a custom rootURL before?

Reply:
Based on the https://emberjs.github.io/rfcs/0095-router-service.html#new-method-url-recognition we should pass the actual URL, but I will test this.

Reply:
Yes, `recognize()` accepts actual URL with root URL ✅.

<img width="1312" alt="Screenshot 2020-10-17 at 17 22 46" src="https://user-images.githubusercontent.com/112557/96346698-827be680-109d-11eb-9c74-c932bc60b0f0.png">